boards/stm32f412gdiscovery/src/main.rs

```diff
@@ -0,0 +1,463 @@
+    // GPIO
+    let gpio = GpioComponent::new(
+        board_kernel,
+        components::gpio_component_helper!(
+            stm32f412g::gpio::Pin,
+            // Arduino like RX/TX
+            0 => stm32f412g::gpio::PinId::PG09.get_pin().as_ref().unwrap(), //D0
+            1 => stm32f412g::gpio::PinId::PG14.get_pin().as_ref().unwrap(), //D1
+            2 => stm32f412g::gpio::PinId::PG13.get_pin().as_ref().unwrap(), //D2
+            3 => stm32f412g::gpio::PinId::PF04.get_pin().as_ref().unwrap(), //D3
+            4 => stm32f412g::gpio::PinId::PG12.get_pin().as_ref().unwrap(), //D4
+            5 => stm32f412g::gpio::PinId::PF10.get_pin().as_ref().unwrap(), //D5
+            6 => stm32f412g::gpio::PinId::PF03.get_pin().as_ref().unwrap(), //D6
+            7 => stm32f412g::gpio::PinId::PG11.get_pin().as_ref().unwrap(), //D7
+            8 => stm32f412g::gpio::PinId::PG10.get_pin().as_ref().unwrap(), //D8
+            9 => stm32f412g::gpio::PinId::PB08.get_pin().as_ref().unwrap(), //D9
+            // SPI Pins
+            10 => stm32f412g::gpio::PinId::PA15.get_pin().as_ref().unwrap(), //D10
+            11 => stm32f412g::gpio::PinId::PA07.get_pin().as_ref().unwrap(),  //D11
+            12 => stm32f412g::gpio::PinId::PA06.get_pin().as_ref().unwrap(),  //D12
+            13 => stm32f412g::gpio::PinId::PA15.get_pin().as_ref().unwrap()  //D13
+
+            // ADC Pins
+            // Enable the to use the ADC pins as GPIO
+            // 14 => stm32f412g::gpio::PinId::PA01.get_pin().as_ref().unwrap(), //A0
+            // 15 => stm32f412g::gpio::PinId::PC01.get_pin().as_ref().unwrap(), //A1
+            // 16 => stm32f412g::gpio::PinId::PC03.get_pin().as_ref().unwrap(), //A2
+            // 17 => stm32f412g::gpio::PinId::PC04.get_pin().as_ref().unwrap(), //A3
+            // 19 => stm32f412g::gpio::PinId::PC05.get_pin().as_ref().unwrap(), //A4
+            // 20 => stm32f412g::gpio::PinId::PB00.get_pin().as_ref().unwrap() //A5
+        ),
+    )
+    .finalize(components::gpio_component_buf!(stm32f412g::gpio::Pin));
+
+    let nucleo_f412g = STM32F412GDiscovery {
+        console: console,
+        ipc: kernel::ipc::IPC::new(board_kernel, &memory_allocation_capability),
+        led: led,
+        button: button,
+        alarm: alarm,
+        gpio: gpio,
+    };
+
+    // // Optional kernel tests
+    // //
+    // // See comment in `boards/imix/src/main.rs`
+    // virtual_uart_rx_test::run_virtual_uart_receive(mux_uart);
+
+    debug!("Initialization complete. Entering main loop");
+
+    extern "C" {
+        /// Beginning of the ROM region containing app images.
+        ///
+        /// This symbol is defined in the linker script.
+        static _sapps: u8;
+
+        /// End of the ROM region containing app images.
+        ///
+        /// This symbol is defined in the linker script.
+        static _eapps: u8;
+    }
+
+    kernel::procs::load_processes(
+        board_kernel,
+        chip,
+        core::slice::from_raw_parts(
```

Comment:
Attn #1962

Reply:
I know about it, but is is not merged yet. I'll update the stm boards as soon as this gets merged. Should I do it differently?

Reply:
Just flagging the (potential) conflict here; whichever goes in first, the other will have to update